### PR TITLE
Major refactor toward data-driven responses and reactions

### PR DESCRIPTION
**Pull request by ScottMansfield**
This PR takes nearly all of the different functions and puts their data into maps that allow the main loop to be much more compact. It places the reactions and responses into several categories:

* Reactions to all messages, including those not directed @gopher

And for messages directed @gopher
* event text -> function, for anything that needs logic
* event text -> text response
* event text prefix -> function, for things that need logic
* event text contains -> reactions
* event text prefix -> reactions

### CODE REVIEW

File: bot/gerrit.go [hunk 1]
```diff
@@ -200,7 +200,12 @@ func (b *Bot) processCLList(lastID int) int {
 	return lastID
 }
 
-func (b *Bot) shareCL(event *slack.MessageEvent, eventText string) {
+func shareCL(b *Bot, event *slack.MessageEvent) {
+	// repeats some earlier work but oh well
+	eventText := strings.Trim(strings.ToLower(event.Text), " \n\r")
+	eventText = b.trimBot(event.Text)
```
Comment (bradleyfalzon) [inline bot/gerrit.go]: Was `event.Text` meant to be `eventText`?

File: bot/bot.go [hunk 2]
```diff
@@ -279,146 +393,66 @@ func (b *Bot) HandleMessage(event *slack.MessageEvent) {
...
-		eventText == "cheers" ||
-		eventText == "hello" {
-		b.reactToEvent(event, "gopher")
-		return
-	}
+	// aliases for the above canned responses
+	if key, ok := botEventTextToResponseAliases[eventText]; ok {
+		if responseLines, ok := botEventTextToResponse[key]; ok {
+			var response string
+			for _, line := range responseLines {
```
Comment (dlsniper) [inline bot/bot.go]: here should be: `strings.Join`

File: bot/bot.go [hunk 3]
```diff
@@ -279,146 +393,66 @@ func (b *Bot) HandleMessage(event *slack.MessageEvent) {
...
-		b.sourceCode(event)
-		return
-	}
-
-	if strings.HasPrefix(eventText, "library for") {
-		b.searchLibrary(event)
+	// Responses that are just a canned string response
+	if responseLines, ok := botEventTextToResponse[eventText]; ok {
+		var response string
+		for _, line := range responseLines {
```
Comment (dlsniper) [inline bot/bot.go]: here should be ` strings.Join `

File: bot/gerrit.go [hunk 4]
```diff
@@ -200,7 +200,12 @@ func (b *Bot) processCLList(lastID int) int {
...
+	eventText = strings.TrimPrefix(eventText, "xkcd:")
```
Comment (dlsniper) [inline bot/gerrit.go]: This should trim the bot ID / bot name, not `xkcd:`

File: bot/gerrit.go (hunk 1 above)
Comment (dlsniper) [inline bot/gerrit.go]: Good catch!

Comment (ScottMansfield) [inline bot/gerrit.go]: whoops, yeah that's a bad copy-paste job

File: bot/gerrit.go (hunk 4 above)
Comment (ScottMansfield) [inline bot/gerrit.go]: I don't think I need the third line actually

File: bot/bot.go (hunk 2 above)
Comment (ScottMansfield) [inline bot/bot.go]: done

File: bot/bot.go (hunk 3 above)
Comment (ScottMansfield) [inline bot/bot.go]: done

File: bot/gerrit.go (hunk 4 above)
Comment (dlsniper) [inline bot/gerrit.go]: Yup

**Pull request merged**